lightstep/resource_metric_dashboard.go

```diff
@@ -890,6 +955,92 @@ func (p *resourceUnifiedDashboardImp) resourceUnifiedDashboardImport(ctx context
 
 }
 
+// buildServiceHealthPanels transforms service_health_panels from the TF resource
+// into panels for the API request.
+func buildServiceHealthPanels(serviceHealthPanelsIn []interface{}) ([]client.Panel, error) {
+	var serviceHealthPanels []client.Panel
+
+	for _, s := range serviceHealthPanelsIn {
+		serviceHealthPanel, ok := s.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("bad format, %v", s)
+		}
+		p := client.Panel{
+			ID:       serviceHealthPanel["id"].(string),
+			Title:    serviceHealthPanel["name"].(string),
+			Position: buildPosition(serviceHealthPanel),
+		}
+		// N.B. panel_options are optional, so we don't return an error if not found
+		if panelOptions, ok := serviceHealthPanel["panel_options"].(map[string]interface{}); ok {
+			displayOptions := make(map[string]interface{})
+			if sortBy, ok := panelOptions["sort_by"]; ok {
+				displayOptions["sort_by"] = sortBy
+			}
+			if sortDirection, ok := panelOptions["sort_direction"]; ok {
+				displayOptions["sort_direction"] = sortDirection
+			}
+			if percentile, ok := panelOptions["percentile"]; ok {
+				displayOptions["percentile"] = percentile
+			}
+			if changeSince, ok := panelOptions["change_since"]; ok {
+				displayOptions["change_since"] = changeSince
+			}
+			p.Body = map[string]interface{}{
+				"display_options": displayOptions,
+			}
+		}
+
+		serviceHealthPanels = append(serviceHealthPanels, p)
+	}
+
+	return serviceHealthPanels, nil
+}
+
+func setServiceHealthPanelResourceData(
+	resource map[string]interface{}, // Terraform resource
+	panel client.Panel, // Panel from the API
+) {
+	resource["name"] = panel.Title
```

Comment:
Maybe a comment here to signal this isn't a typo and we intentionally need to use `name` vs `title` in this case?